src/extension.c

```diff
@@ -281,6 +283,21 @@ ts_extension_is_loaded(void)
 			 * that, for example, the catalog does not go looking for things
 			 * that aren't yet there.
 			 */
+			if (extstate == EXTENSION_STATE_TRANSITIONING)
+			{
+				/* when we are updating the extension, we execute
+				 * scripts in post_update.sql after setting up the
+				 * the dependencies. At this stage, TS
+				 * specific functionality is permitted as we now have
+				 * all catalogs and functions in place
+				 */
+				const char *update_script_stage =
+					GetConfigOption(TS_UPDATE_SCRIPT_CONFIG_VAR, true, false);
+				if (update_script_stage &&
+					(strncmp(update_script_stage, POST_UPDATE, strlen(POST_UPDATE)) == 0) &&
+					(strlen(POST_UPDATE) == strlen(update_script_stage)))
```

Comment:
```suggestion
					(strcmp(update_script_stage, POST_UPDATE) == 0))
```

Reply:
Since `POST_UPDATE` is a string literal, there is no risk of an unterminated string. Here's how `strcmp` is implemented in glibc:

```c
int
STRCMP (const char *p1, const char *p2)
{
  const unsigned char *s1 = (const unsigned char *) p1;
  const unsigned char *s2 = (const unsigned char *) p2;
  unsigned char c1, c2;

  do
    {
      c1 = (unsigned char) *s1++;
      c2 = (unsigned char) *s2++;
      if (c1 == '\0')
        return c1 - c2;
    }
  while (c1 == c2);

  return c1 - c2;
}
```